tests/UnitTests/Foundation/NSObject_NSKeyValueOrderedSetAdaptersTests.mm

```diff
@@ -0,0 +1,265 @@
+        _internal_immutableThroughKVC = [value copy];
+        immutableThroughKVCOrderedSet = YES;
+        return;
+    }
+    return [super setValue:value forKey:key];
+}
+
+- (BOOL)immutableThroughKVCOrderedSet {
+    return immutableThroughKVCOrderedSet;
+}
+@end
+
+TEST(NSObject, KVCOrderedSetMutableAdapters_CompliantIndexedAccessors) {
+    TestOrderedSetAdapterObject* testObject = [[[TestOrderedSetAdapterObject alloc] init] autorelease];
+    NSMutableOrderedSet* fakeMutableCollection = [testObject mutableOrderedSetValueForKey:@"fakeMutableCollection"];
+    EXPECT_TRUE([fakeMutableCollection isKindOfClass:[NSMutableOrderedSet class]]);
+
+    EXPECT_EQ(10, [fakeMutableCollection count]);
+    EXPECT_OBJCEQ(@5, [fakeMutableCollection objectAtIndex:5]);
+}
+
+TEST(NSObject, KVCOrderedSetMutableAdapters_InsertObject) {
+    TestOrderedSetAdapterObject* testObject = [[[TestOrderedSetAdapterObject alloc] init] autorelease];
+    NSMutableOrderedSet* fakeCollection = [testObject mutableOrderedSetValueForKey:@"fakeMutableCollection"];
+    EXPECT_NO_THROW([fakeCollection insertObject:@"TEST" atIndex:0]);
+    EXPECT_TRUE([testObject fakeMutableInserted]);
+}
+
+TEST(NSObject, KVCOrderedSetMutableAdapters_RemoveObject) {
+    TestOrderedSetAdapterObject* testObject = [[[TestOrderedSetAdapterObject alloc] init] autorelease];
+    id fakeCollection = [testObject mutableOrderedSetValueForKey:@"fakeMutableCollection"];
+    EXPECT_NO_THROW([fakeCollection removeObjectAtIndex:0]);
+    EXPECT_TRUE([testObject fakeMutableRemoved]);
+}
+
+TEST(NSObject, KVCOrderedSetMutableAdapters_InsertObjects) {
+    TestOrderedSetAdapterObject* testObject = [[[TestOrderedSetAdapterObject alloc] init] autorelease];
+    id fakeCollection = [testObject mutableOrderedSetValueForKey:@"fakeMutableCollection"];
+    EXPECT_NO_THROW([fakeCollection insertObjects:@[ @1 ] atIndexes:[NSIndexSet indexSetWithIndex:5]]);
+    EXPECT_TRUE([testObject fakeMutableInsertedAtIndexes]);
+}
+
+TEST(NSObject, KVCOrderedSetMutableAdapters_RemoveObjects) {
+    TestOrderedSetAdapterObject* testObject = [[[TestOrderedSetAdapterObject alloc] init] autorelease];
+    id fakeCollection = [testObject mutableOrderedSetValueForKey:@"fakeMutableCollection"];
+    EXPECT_NO_THROW([fakeCollection removeObjectsAtIndexes:[NSIndexSet indexSetWithIndex:5]]);
+    EXPECT_TRUE([testObject fakeMutableRemovedAtIndexes]);
+}
+
+TEST(NSObject, KVCOrderedSetMutableAdapters_ImmutableWithBasicAccessors) {
+    TestOrderedSetAdapterObject* testObject = [[[TestOrderedSetAdapterObject alloc] init] autorelease];
+    id mutableImmutable = [testObject mutableOrderedSetValueForKey:@"immutableWithAccessors"];
+    EXPECT_TRUE([mutableImmutable isKindOfClass:[NSMutableOrderedSet class]]);
+    EXPECT_NO_THROW([mutableImmutable addObject:@"Whatever"]);
+
+    ASSERT_TRUE([testObject immutableWithAccessorsOrderedSet]);
+    EXPECT_TRUE([[testObject immutableWithAccessors] containsObject:@"Whatever"]);
+}
```

Comment:
They'd be parameterized on the selectors, and the reference platform is not having it. #ByDesign